mobile/library/java/io/envoyproxy/envoymobile/engine/JvmFilterContext.java

```diff
@@ -66,11 +66,8 @@ public Object onRequestHeaders(long headerCount, boolean endStream, long[] strea
    * @return Object[],   pair of HTTP filter status and optional modified data.
    */
   public Object onRequestData(ByteBuffer data, boolean endStream, long[] streamIntel) {
-    // Create a copy of the `data` because the `data` uses direct `ByteBuffer` and the `data` will
-    // be destroyed after calling this callback.
-    ByteBuffer copiedData = ByteBuffers.copy(data);
     return toJniFilterDataStatus(
-        filter.onRequestData(copiedData, endStream, new EnvoyStreamIntelImpl(streamIntel)));
```

Comment:
question: we weren't running this in a separate thread from the looks of it, so why did we need `copiedData`?

Reply:
Yeah. I did this initially to be consistent with the callbacks API, but now that we have documented what the behavior is going to be with regards to the `ByteBuffer`, we can simply remove this logic.